lasio/reader.py

```diff
@@ -68,28 +66,61 @@ def open_file(file_ref, encoding=None, encoding_errors='replace',
         An open file-like object ready for reading from.
 
     '''
-    if isinstance(file_ref, str):
-        lines = file_ref.splitlines()
-        if len(lines) == 1:  # File name
-            if URL_REGEXP.match(file_ref):
-                try:
-                    import urllib2
-                    file_ref = urllib2.urlopen(file_ref)
-                except ImportError:
-                    import urllib.request
-                    response = urllib.request.urlopen(file_ref)
-                    enc = response.headers.get_content_charset('utf-8')
-                    file_ref = StringIO(response.read().decode(enc))
-            else:  # filename
-                data = get_unicode_from_filename(
-                    file_ref, encoding, encoding_errors, autodetect_encoding,
-                    autodetect_encoding_chars)
-                file_ref = StringIO(data)
-        else:
-            file_ref = StringIO('\n'.join(lines))
+    if isinstance(file_ref, str): # file_ref != file-like object, so what is it?
+        file_ref = file_ref.splitlines()[0]
+        if URL_REGEXP.match(file_ref): # file_ref == URI
+            try:
+                import urllib2
+                file_ref = urllib2.urlopen(file_ref)
+            except ImportError:
+                import urllib.request
+                response = urllib.request.urlopen(file_ref)
+                enc = response.headers.get_content_charset('utf-8')
+                file_ref = StringIO(response.read().decode(enc))
+        else:  # file_ref == filename
+            file_ref = codecs.open(file_ref, 'r', encoding, encoding_errors)
+    # else: file_ref is already a file-like object.
     return file_ref
 
 
+def read(file_ref, null_subs, **kwargs):
+    sections = OrderedDict()
+    sect_lines = []
+    sect_title_line = None
+    file_ref = open_file(file_ref, **kwargs)
+    for line in file_ref:
+        line = line.strip()
+        if line.startswith('~A'):
+            # We have finished looking at the metadata and need
+            # to start reading numerical data.
```

Comment:
If a user were interested in _only_ the header data in a LAS file, this could be a good place to add a `header_only` option. If `True` it would not read in the log data. In other words, could be something like ```if not header_only: 
    do stuff with ~A section```
Not having to read in all the log data would significantly speed things up if the user is only interested in header data. Just a thought. Cheers.